delfin/api/v1/storages.py

```diff
@@ -163,19 +165,17 @@ def sync(self, req, id):
                 subclass.__module__ + '.' + subclass.__name__)
 
     def _storage_exist(self, context, access_info):
-        access_info_dict = copy.deepcopy(access_info)
-
-        # Remove unrelated query fields
-        unrelated_fields = ['username', 'password']
-        for access in constants.ACCESS_TYPE:
-            if access_info_dict.get(access):
-                for key in unrelated_fields:
-                    access_info_dict[access].pop(key)
+        access_info_filter = {'vendor': access_info['vendor'],
+                              'model': access_info['model']}
 
         # Check if storage is registered
         access_info_list = db.access_info_get_all(context,
-                                                  filters=access_info_dict)
+                                                  filters=access_info_filter)
+
         for _access_info in access_info_list:
+            # Check whether access_info_feature matches the exist _access_info
+            if not _is_same_access_info(access_info, dict(_access_info)):
```

Comment:
this function currently checks whether ports given in acces_info are same? I hope we cant decide based on port, for example 2 or more vmax can be discovered from same unisphereIP:port with different extra_attribute {array_id = <array to be discovered>}

Reply:
Now we'll check whether IP/port pair is same to decide whether two access_info are same

Reply:
If we do this , it will be regression to VMAX registration. Now we can register different VMAX arrays from same unisphere by providing array_id as extra attribute.. but with this code that feature will break and we can add only one array from unisphere.  @wisererik @kumarashit your views please..

Reply:
Its incorrect to register only one array of the Unisphere. Either we keep three tuple (ip, port, storageid) or register all the arrays of the Unisphere in *one go*

Reply:
one unsipshere can represent mutiple vmax arrays..  This is why we introduced extra  attibutes in resgitration body ( in vmax it is aray_id) to specify which array to discover.  With this change that feature will break,  If user wants to dicover more than one array from a unisphere he has no way . @skdwriting Please suggest..